servicetalk-tcp-netty-internal/src/main/java/io/servicetalk/tcp/netty/internal/ReadOnlyTcpClientConfig.java

```diff
@@ -68,4 +80,58 @@ public String preferredAlpnProtocol() {
     public ClientSslConfig sslConfig() {
         return sslConfig;
     }
+
+    /**
+     * Create a new {@link ReadOnlyTcpClientConfig} replacing {@link ClientSslConfig#peerHost()},
+     * {@link ClientSslConfig#peerPort()}, and {@link ClientSslConfig#sniHostname()}.
+     * @param peerHost The new value for {@link ClientSslConfig#peerHost()}.
+     * @param peerPort The new value for {@link ClientSslConfig#peerPort()}.
+     * @param sniHostname The new value for {@link ClientSslConfig#sniHostname()}.
+     * @param hostnameVerificationAlgorithm The new value for {@link ClientSslConfig#hostnameVerificationAlgorithm()}.
+     * @return a new {@link ReadOnlyTcpClientConfig} replacing {@link ClientSslConfig#peerHost()} and
+     * {@link ClientSslConfig#peerPort()}.
+     */
+    public ReadOnlyTcpClientConfig withSslConfigPeerHost(String peerHost, int peerPort, @Nullable String sniHostname,
+                                                         @Nullable final String hostnameVerificationAlgorithm) {
+        return new ReadOnlyTcpClientConfig(this, peerHost, peerPort, sniHostname, hostnameVerificationAlgorithm);
+    }
+
+    private static final class PeerHostNameOverrideClientSslConfig extends DelegatingClientSslConfig {
+        private final String peerHost;
+        private final int peerPort;
+        @Nullable
+        private final String sniHostname;
+        @Nullable
+        private final String hostnameVerificationAlgorithm;
+
+        PeerHostNameOverrideClientSslConfig(final ClientSslConfig delegate, final String peerHost, final int peerPort,
+                                            @Nullable final String sniHostname,
+                                            @Nullable final String hostnameVerificationAlgorithm) {
+            super(delegate);
+            this.peerHost = peerHost;
+            this.peerPort = peerPort;
+            this.sniHostname = sniHostname;
+            this.hostnameVerificationAlgorithm = hostnameVerificationAlgorithm;
+        }
+
+        @Override
+        public String peerHost() {
+            return peerHost;
+        }
+
+        @Override
+        public int peerPort() {
+            return peerPort;
+        }
+
+        @Override
+        public String sniHostname() {
```

Comment:
Not related to this PR, but something to consider: is it worth changing a return type of this method to `javax.net.ssl.SNIHostName` instead of a `String`? It will eliminate a case when users passed a `String` that won't pass `SNIHostName` validation.

Reply:
good point. will consider for followup PR.